test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -374,6 +375,15 @@ private static void setTestSysProps() {
         // We have to disable setting the number of available processors as tests in the same JVM randomize processors and will step on each
         // other if we allow them to set the number of available processors as it's set-once in Netty.
         System.setProperty("es.set.netty.runtime.available.processors", "false");
+
+        // sometimes use the java.time date formatters
+        // we can't use randomBoolean here, the random context isn't set properly
+        // so read it directly from the test seed in an unfortunately hacky way
+        String testSeed = System.getProperty("tests.seed", "0");
```

Comment:
The seed should already be available in this test context? Just use randomBoolean()

Reply:
Unfortunately not, that throws in `RandomizedContext.context` saying there's no context info for the thread

Reply:
Ah, I think this is due to limitations in JUnit 4. Because the Runner is passed an already initialized suite Class, so this static init runs before the Runner can actually setup the context.